client/controllers/curatorInbox/addToEvent.coffee

```diff
@@ -24,15 +25,19 @@ Template.addToEvent.onRendered ->
       placeholder: 'Search for an Event...'
       minimumInputLength: 0
 
-    $(document).on 'click', '.add-new-event', (event) ->
+    $(document).on 'click', ".add-new-event-#{instanceData.objNameToAssociate}", (event) ->
```

Comment:
Seems like the issue was related to the select2 related DOM elements being detached from Blaze or the template. I tried using a template click handler but the event never fires.